statsmodels/discrete/count_model.py

```diff
@@ -57,12 +58,16 @@ def __init__(self, endog, exog, exog_infl=None, offset=None,
         else:
             self.k_exog = exog.shape[1]
 
+        self.infl = inflation
         if inflation == 'logit':
             self.model_infl = Logit(np.zeros(self.exog_infl.shape[0]),
                                     self.exog_infl)
+            self._hessian_inflate = self._hessian_logit
         elif inflation == 'probit':
             self.model_infl = Probit(np.zeros(self.exog_infl.shape[0]),
                                     self.exog_infl)
+            self._hessian_inflate = self._hessian_probit
```

Comment:
this might need refactoring, I think this will be an instance method that prevents pickling
can be done after we add generic pickling tests, changing this later shouldn't be difficult
(I'm not familiar enough with the overall structure of methods and attached models to tell right now)